src/paste.rs

```diff
@@ -116,28 +116,61 @@ impl Paste {
             Some(v) => v.to_string(),
             None => String::from("file"),
         };
+        let mut input = file_name.to_string();
+
         let mut path = self
             .type_
             .get_path(&config.server.upload_path)
             .join(file_name);
-        match path.clone().extension() {
-            Some(extension) => {
-                if let Some(file_name) = config.paste.random_url.generate() {
-                    path.set_file_name(file_name);
-                    path.set_extension(extension);
-                }
+
+        // set_file_name does not accept names that contain a . unless it is the first character (e.g. "file.name" not ok, ".file" ok)
+        // as a workaround we use set_extension, because it can set multiple extensions (e.g. "tar.gz")
+        if input == "." {
+            input = "file".to_string();
+        }
+
+        let mut v: Vec<&str> = input.split('.').collect();
+
+        let mut filename;
+        let mut dotfile = false;
+        let first = v[0];
+        if !first.is_empty() {
+            filename = v[0].to_string();
+        } else {
+            filename = format!(".{}", v[1]);
+            dotfile = true;
+        }
+
+        let mut extension;
+        let l = v.len();
+        if l > 1 {
+            v.remove(0);
+            if dotfile {
+                v.remove(0);
```

Comment:
I have trouble understanding this code. Can you maybe avoid single line declarations (if possible) and remove the unnecessary line breaks (and maybe use comments)? Frankly, ChatGPT might really help refactor this.

Reply:
I have never used ChatGPT and I am not willing to use it. I can explain the code though:

```rust
    // This is just a safeguard, in case a filename is sent that is just a dot.
    // It should never happen, but who knows. Paranoia took over, which is why I added this.
    if input == "." {
        input = "file".to_string();
    }

    // Split the string into an array
    let mut v: Vec<&str> = input.split('.').collect();

    let mut filename;
    let mut dotfile = false;
    // This the first element of the array 
    let first = v[0];
    if !first.is_empty() {
        filename = v[0].to_string();
    } else {
        // If the first array element is empty, it means the file started with a .
        filename = format!(".{}", v[1]);
        // Therefore the index shifts one to the right in the array for the rest of the string
        dotfile = true;
    }

    let mut extension;
    let l = v.len();
    if l > 1 {
        // To get the rest (the extension), we have to remove the first element of the array, which is the filename
        v.remove(0);
        if dotfile {
            // as mentioned above, if the file starts with a ., we have to remove another element because the filename is the second element in the array
            v.remove(0);
```

I am sorry that I didn't add any comments. I thought that by reading the code it was clear what the purpose was. Let me know what you think of the comments.
If they are ok, I'll add them and fix the line breaks and the unnecessary `let first = v[0];`

Reply:
I think I can improve the comments. I'll do that later today.

Reply:
I think that's done.